### PR TITLE
feat: add a wiring stub for gossip

### LINKED ISSUE #11169: Create wiring stub for gossip
Create a pass-through stub that allows gossip to put its events onto a proper wire. Will cause gossip to show up in the wiring framework.

We eventually should move gossip entirely into the framework. This is a good incremental step that will clean up our wiring diagram.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #11169 
[wiring diagram](https://mermaid.ink/svg/Zmxvd2NoYXJ0IExSCnYwKCgiQCIpKQpzdWJncmFwaCB2MVsiQ29uc2Vuc3VzIFBpcGVsaW5lIl0Kc3ViZ3JhcGggdjJbIkxpbmtlZCBFdmVudCBJbnRha2UiXQp2M3t7ImdldE5vbkFuY2llbnRFdmVudFdpbmRvdyJ9fQp2NFsibGlua2VkRXZlbnRJbnRha2U8YnIgLz7imIUiXQp2NXt7ImxpbmtlZEV2ZW50SW50YWtlU3BsaXR0ZXIifX0KZW5kCnY2WyJpbk9yZGVyTGlua2VyPGJyIC8+ypgiXQplbmQKc3ViZ3JhcGggdjdbIkdvc3NpcCJdCnY4Wy8iZ29zc2lwIi9dCnY5WyJzaGFkb3dncmFwaDxiciAvPmfKmCJdCmVuZApzdWJncmFwaCB2MTBbIkludGFrZSBQaXBlbGluZSJdCnN1YmdyYXBoIHYxMVsiRXZlbnQgSGFzaGluZyJdCnYxMltbImV2ZW50SGFzaGVyIl1dCnYxM1sicG9zdEhhc2hDb2xsZWN0b3IiXQplbmQKc3ViZ3JhcGggdjE0WyJFdmVudCBWYWxpZGF0aW9uIl0KdjE1WyJldmVudERlZHVwbGljYXRvcjxiciAvPsqYIl0KdjE2WyJldmVudFNpZ25hdHVyZVZhbGlkYXRvcjxiciAvPsqYIl0KdjE3WyJpbnRlcm5hbEV2ZW50VmFsaWRhdG9yPGJyIC8+4oCgIl0KZW5kCnYxOFsiT3JwaGFuIEJ1ZmZlcjxiciAvPsqYIl0KZW5kCnN1YmdyYXBoIHYxOVsiUENFUyJdCnYyMFsvImV2ZW50RHVyYWJpbGl0eU5leHVzIi9dCnYyMVsvInBjZXNTZXF1ZW5jZXIiL10KdjIyWyJwY2VzV3JpdGVyPGJyIC8+QMqYIl0KZW5kCnN1YmdyYXBoIHYyM1siU3RhdGUgRmlsZSBNYW5hZ2VtZW50Il0KdjI0e3siZXh0cmFjdE9sZGVzdE1pbmltdW1HZW5lcmF0aW9uT25EaXNrIn19CnYyNXt7InNhdmVUb0Rpc2tGaWx0ZXIifX0KdjI2WyJzaWduZWRTdGF0ZUZpbGVNYW5hZ2VyIl0KdjI3e3sidG9TdGF0ZVdyaXR0ZW5Ub0Rpc2tBY3Rpb24ifX0KZW5kCnN1YmdyYXBoIHYyOFsiU3RhdGUgU2lnbmF0dXJlIENvbGxlY3Rpb24iXQp2Mjl7eyJhbGxTdGF0ZXNSZXNlcnZlciJ9fQp2MzB7eyJjb21wbGV0ZVN0YXRlRmlsdGVyIn19CnYzMXt7ImNvbXBsZXRlU3RhdGVzUmVzZXJ2ZXIifX0KdjMye3siZXh0cmFjdENvbnNlbnN1c1NpZ25hdHVyZVRyYW5zYWN0aW9ucyJ9fQp2MzN7eyJleHRyYWN0UHJlY29uc2Vuc3VzU2lnbmF0dXJlVHJhbnNhY3Rpb25zIn19CnYzNHt7InJlc2VydmVkU3RhdGVTcGxpdHRlciJ9fQp2MzVbInN0YXRlU2lnbmF0dXJlQ29sbGVjdG9yIl0KZW5kCnYzNltbImFwcGxpY2F0aW9uVHJhbnNhY3Rpb25QcmVoYW5kbGVyIl1dCnYzN1siZXZlbnRDcmVhdGlvbk1hbmFnZXI8YnIgLz7KmOKZoSJdCnYzOCgoImciKSkKdjM5WyJoZWFydGJlYXQiXQp2NDBbLyJwY2VzUmVwbGF5ZXIiL10KdjQxWyJzdGF0ZVNpZ25lciJdCnY0MigoIsqYIikpCnY0MygoIuKAoCIpKQp2NDQoKCLimIUiKSkKdjQ1KCgi4pmhIikpCnYxOCAtLSAicHJlY29uc2Vuc3VzIGV2ZW50cyIgLS0+IHYzNgp2MTggLS0gInBvc3NpYmxlIHBhcmVudHMiIC0tPiB2MzcKdjE4IC0tICJwcmVjb25zZW5zdXMgZXZlbnRzIiAtLT4gdjMzCnYxOCAtLSAiZXZlbnRzIHRvIHNlcXVlbmNlIiAtLT4gdjIxCnYyOSAtLSAic3RhdGVzIiAtLT4gdjMwCnYyOSAtLSAic3RhdGVzIiAtLT4gdjI1CnYzNiAtLiAiZnV0dXJlcyIgLi1vIHY0NAp2MzAgLS0+IHYzMQp2MzcgLS4gIm5vbi12YWxpZGF0ZWQgZXZlbnRzIiAuLT4gdjQzCnYxNSAtLSAiZXZlbnRzIHdpdGggdW52YWxpZGF0ZWQgc2lnbmF0dXJlcyIgLS0+IHYxNgp2MTIgLS0gImhhc2hlZCBldmVudHMiIC0tPiB2MTMKdjE2IC0tICJ1bm9yZGVyZWQgZXZlbnRzIiAtLT4gdjE4CnYzMiAtLSAiY29uc2Vuc3VzIHNpZ25hdHVyZSB0cmFuc2FjdGlvbnMiIC0tPiB2MzUKdjI0IC0tICJtaW5pbXVtIGlkZW50aWZpZXIgdG8gc3RvcmUiIC0tPiB2MjIKdjMzIC0tICJwcmVjb25zZW5zdXMgc2lnbmF0dXJlIHRyYW5zYWN0aW9ucyIgLS0+IHYzNQp2MyAtLiAibm9uLWFuY2llbnQgZXZlbnQgd2luZG93IiAuLT4gdjQyCnY4IC0tICJldmVudHMgdG8gaGFzaCIgLS0+IHYxMgp2MzkgLS0gImhlYXJ0YmVhdCIgLS0+IHY0NQp2NiAtLSAiZXZlbnRzIHRvIGdvc3NpcCIgLS0+IHYzOAp2NiAtLSAibGlua2VkIGV2ZW50cyIgLS0+IHY0CnYxNyAtLSAibm9uLWRlZHVwbGljYXRlZCBldmVudHMiIC0tPiB2MTUKdjQgLS0gInJvdW5kIGxpc3RzIiAtLT4gdjUKdjQgLS0gImZsdXNoIHJlcXVlc3QiIC0tPiB2MjIKdjUgLS0gInJvdW5kcyIgLS0+IHYzCnY0MCAtLSAiZG9uZSBzdHJlYW1pbmcgcGNlcyIgLS0+IHYwCnY0MCAtLSAiZXZlbnRzIHRvIGhhc2giIC0tPiB2MTIKdjIxIC0tICJ1bmxpbmtlZCBldmVudHMiIC0tPiB2Ngp2MjEgLS0gImV2ZW50cyB0byB3cml0ZSIgLS0+IHYyMgp2MjIgLS0gImxhdGVzdCBkdXJhYmxlIHNlcXVlbmNlIG51bWJlciIgLS0+IHYyMAp2MTMgLS0gIm5vbi12YWxpZGF0ZWQgZXZlbnRzIiAtLT4gdjE3CnYzNCAtLT4gdjI5CnYyNSAtLSAic3RhdGVzIHRvIHNhdmUiIC0tPiB2MjYKdjI2IC0tICJzdGF0ZSBzYXZpbmcgcmVzdWx0IiAtLT4gdjI0CnYyNiAtLSAic3RhdGUgc2F2aW5nIHJlc3VsdCIgLS0+IHYyNwp2MzUgLS0gInJlc2VydmVkIHN0YXRlIGxpc3RzIiAtLT4gdjM0CmNsYXNzRGVmIHMwIGZpbGw6I2Y4OCxzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYwLHYzOCx2NDIsdjQzLHY0NCx2NDUgczAKY2xhc3NEZWYgczEgZmlsbDojOUNGLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjEsdjEwLHYxOSx2MjMsdjI4LHY3IHMxCmNsYXNzRGVmIHMyIGZpbGw6I0JFRixzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYxMSx2MTQsdjIgczIKY2xhc3NEZWYgczMgZmlsbDojY2NjLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjIwLHYyMSx2MjQsdjI1LHYyNyx2MjksdjMsdjMwLHYzMSx2MzIsdjMzLHYzNCx2NDAsdjUsdjggczMKY2xhc3NEZWYgczQgZmlsbDojZmY5LHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjEyLHYxMyx2MTUsdjE2LHYxNyx2MTgsdjIyLHYyNix2MzUsdjM2LHYzNyx2MzksdjQsdjQxLHY2LHY5IHM0Cg==?bgColor=e8e8e8)

### CODE REVIEW

**Review by edward-swirldslabs — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java [hunk 1]
```diff
@@ -323,15 +328,12 @@
 
     /**
      * Get the input wire for the hasher.
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:330]: Out of date method description

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/GossipWiring.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ */
+public record GossipWiring(@NonNull InputWire<GossipEvent> eventInput, @NonNull OutputWire<GossipEvent> eventOutput) {
+
+    /**
+     * Create a new instance of {@link GossipWiring}.
+     *
+     * @return the new instance
+     */
+    @NonNull
+    public static GossipWiring create(@NonNull final WiringModel model) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/GossipWiring.java:42]: missing param doc for `model`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/GossipWiring.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        final TaskScheduler<GossipEvent> scheduler = model.schedulerBuilder("gossip")
+                .withType(TaskSchedulerType.DIRECT)
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/GossipWiring.java:44]: Any reason not to be stateless?

**Review by alittley — APPROVED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/GossipWiring.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/GossipWiring.java:42]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:330]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/GossipWiring.java (hunk 3 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/components/GossipWiring.java:44]: Changed to stateless.

**Review by alittley — DISMISSED** [review]

**Review by alittley — APPROVED** [review]

**Comment by cody-littley** [comment]
JRS panel failed due to a well understood reason (prior PR broke it). Merging despite JRS panel.

**Pull request merged**